odoo/models.py

```diff
@@ -4162,15 +4159,6 @@ def apply_rule(clauses, params, tables, parent_model=None):
                     if table not in query.tables:
                         query.tables.append(table)
 
-        if self._transient:
-            # One single implicit access rule for transient models: owner only!
-            # This is ok because we assert that TransientModels always have
-            # log_access enabled, so that 'create_uid' is always there.
-            domain = [('create_uid', '=', self._uid)]
```

Comment:
@mart-e 
I don't really know this part of Odoo (so, sorry if I miss-understood something), but I have a question

If I understand correctly, everybody will be able to read the transient model of other users.
What if I change my password ? Will other user will be able to read what I entered before it's cleaned by the autovacuum ?

Reply:
(or other sensitive wizard, can be interesting to force the domain on all transient model)

Reply:
In theory yes but no issue for the change password as
- I added an ir.rule to replicate this in this PR
- When making the change_password action, [we clear the value](https://github.com/odoo/odoo/blob/cec3a12a37468311a52dbaccbf9376d761b74c34/odoo/addons/base/models/res_users.py#L1397)

I added it on a few other models like mail.compose.message and payment related stuff. However, we have to make sure we don't miss another one.

Reply:
@mart-e why do not force the domain on all transient model ? There'are some situation when we need to read the model created by another user ?

Reply:
I agree with you, it's still an ongoing discussion. On the one hand, we rarely need it and it is overkill to always filter. On the other hand, we don't need to read other user's transient records.

Reply:
@std-odoo after discussion with Olivier and Raphael, we will keep it this way (remove the tacit ir.rule). This way, the transient are just like other models, you have to define ir.rule if it contains sensible information. We fully remove the "blackmagic" of automatic ir.rule that nobody really knows about it.
Hopefully the warning about ir.model.access will help to change how dev consider transient models.

Reply:
@mart-e Ok :) maybe we should send a email to all the developers, and said that for "sensible" transient models, they must add a domain on it :) (to avoid future issue)

Reply:
I still think it would be better with `('create_uid', '=', self._uid)` though. Is there a real cost to leave it?
Having to add manually more restrictive rules for sensitive data could lead to forgotten ir.rules...